--- a/glib/src/glib_functions.defs
+++ b/glib/src/glib_functions.defs
@@ -7672,7 +7672,7 @@
   (c-name "g_io_channel_write_chars")
   (return-type "GIOStatus")
   (parameters
+    '("const-gchar*" "buf")
-    '("gchar*" "buf")
     '("gssize" "count")
     '("gsize*" "bytes_written")
     '("GError**" "error")

@@ -7961,7 +7961,7 @@
   (return-type "gboolean")
   (parameters
     '("gchar*" "file")
+    '("const-gchar**" "search_dirs")
-    '("gchar**" "search_dirs")
     '("gchar**" "full_path")
     '("GKeyFileFlags" "flags")
     '("GError**" "error")

@@ -10330,7 +10330,7 @@
   (c-name "g_regex_replace")
   (return-type "gchar*")
   (parameters
+    '("const-gchar*" "string")
-    '("gchar*" "string")
     '("gssize" "string_len")
     '("gint" "start_position")
     '("const-gchar*" "replacement")

@@ -10359,7 +10359,7 @@
   (c-name "g_regex_replace_literal")
   (return-type "gchar*")
   (parameters
+    '("const-gchar*" "string")
-    '("gchar*" "string")
     '("gssize" "string_len")
     '("gint" "start_position")
     '("const-gchar*" "replacement")

@@ -10383,7 +10383,7 @@
   (c-name "g_regex_split_full")
   (return-type "gchar**")
   (parameters
+    '("const-gchar*" "string")
-    '("gchar*" "string")
     '("gssize" "string_len")
     '("gint" "start_position")
     '("GRegexMatchFlags" "match_options")
