--- glib_functions.defs	2012-02-28 10:09:30.000000000 -0500
+++ glib_functions.defs.new	2012-02-28 10:08:25.000000000 -0500
@@ -2563,7 +2563,7 @@
   )
 )
 
-(define-method 
+(define-method iconv
   (of-object "GIConv")
   (c-name "g_iconv")
   (return-type "gsize")
@@ -12346,13 +12346,9 @@
   (return-type "const-gchar*-const*")
 )
 
-(define-function g_win32_get_system_data_dirs_for_module
-  (c-name "g_win32_get_system_data_dirs_for_module")
-  (return-type "const-gchar*-const*")
-  (parameters
-    '("void-(*" "address_of_function")
-  )
-)
+;; the g_win32_get_system_data_dirs_for_module() function is removed because it
+;; has a function pointer parameter and that is not handled correctly by
+;; h2def.py.
 
 (define-function g_get_user_runtime_dir
   (c-name "g_get_user_runtime_dir")
@@ -12440,13 +12436,8 @@
   )
 )
 
-(define-function atexit
-  (c-name "atexit")
-  (return-type "int")
-  (parameters
-    '("void" "(*")
-  )
-)
+;; the atexit() function is removed because it has a function pointer parameter
+;; and that is not handled correctly by h2def.py.
 
 (define-function g_find_program_in_path
   (c-name "g_find_program_in_path")
