--- a/gio_methods.defs
+++ b/gio_methods.defs
@@ -1551,13 +1551,13 @@
 (define-method get_cwd
   (of-object "GApplicationCommandLine")
   (c-name "g_application_command_line_get_cwd")
-  (return-type "gchar*")
+  (return-type "const-gchar*")
 )
 
 (define-method get_environ
   (of-object "GApplicationCommandLine")
   (c-name "g_application_command_line_get_environ")
-  (return-type "gchar**")
+  (return-type "const-gchar*-const*")
 )
 
 (define-method get_exit_status

@@ -1705,7 +1705,7 @@
 (define-method peek_buffer
   (of-object "GBufferedInputStream")
   (c-name "g_buffered_input_stream_peek_buffer")
-  (return-type "void*")
+  (return-type "const-void*")
   (parameters
     '("gsize*" "count")
   )

@@ -4317,7 +4317,7 @@
 (define-method get_filename
   (of-object "GDesktopAppInfo")
   (c-name "g_desktop_app_info_get_filename")
-  (return-type "char*")
+  (return-type "const-char*")
 )
 
 (define-method get_generic_name

@@ -4406,7 +4406,7 @@
 (define-method list_actions
   (of-object "GDesktopAppInfo")
   (c-name "g_desktop_app_info_list_actions")
-  (return-type "gchar**")
+  (return-type "const-gchar*-const*")
 )
 
 (define-function g_emblem_new

@@ -4970,7 +4970,7 @@
 (define-method get_name
   (of-object "GFileInfo")
   (c-name "g_file_info_get_name")
-  (return-type "char*")
+  (return-type "const-char*")
 )
 
 (define-method get_size

@@ -5713,7 +5713,7 @@
   (c-name "g_inet_address_new_from_bytes")
   (return-type "GInetAddress*")
   (parameters
-    '("guint8*" "bytes")
+    '("const-guint8*" "bytes")
     '("GSocketFamily" "family")
   )
 )

@@ -6243,7 +6243,7 @@
   (c-name "g_memory_input_stream_add_data")
   (return-type "void")
   (parameters
-    '("void*" "data")
+    '("const-void*" "data")
     '("gssize" "len")
     '("GDestroyNotify" "destroy")
   )

@@ -7217,7 +7217,7 @@
   (c-name "g_output_stream_write")
   (return-type "gssize")
   (parameters
-    '("void*" "buffer")
+    '("const-void*" "buffer")
     '("gsize" "count")
     '("GCancellable*" "cancellable")
     '("GError**" "error")

@@ -7229,7 +7229,7 @@
   (c-name "g_output_stream_write_all")
   (return-type "gboolean")
   (parameters
-    '("void*" "buffer")
+    '("const-void*" "buffer")
     '("gsize" "count")
     '("gsize*" "bytes_written")
     '("GCancellable*" "cancellable")

@@ -8157,7 +8157,7 @@
   (return-type "gboolean")
   (parameters
     '("const-gchar*" "key")
-    '("gchar**" "value")
+    '("const-gchar*-const*" "value")
   )
 )
 
@@ -9047,7 +9047,7 @@
   (c-name "g_socket_send")
   (return-type "gssize")
   (parameters
-    '("gchar*" "buffer")
+    '("const-gchar*" "buffer")
     '("gsize" "size")
     '("GCancellable*" "cancellable")
     '("GError**" "error")

@@ -9089,7 +9089,7 @@
   (return-type "gssize")
   (parameters
     '("GSocketAddress*" "address")
-    '("gchar*" "buffer")
+    '("const-gchar*" "buffer")
     '("gsize" "size")
     '("GCancellable*" "cancellable")
     '("GError**" "error")

@@ -12220,7 +12220,7 @@
 (define-method get_commandline
   (of-object "GAppInfo")
   (c-name "g_app_info_get_commandline")
-  (return-type "char*")
+  (return-type "const-char*")
 )
 
 (define-method get_description

@@ -12238,7 +12238,7 @@
 (define-method get_executable
   (of-object "GAppInfo")
   (c-name "g_app_info_get_executable")
-  (return-type "char*")
+  (return-type "const-char*")
 )
 
 (define-method get_icon

@@ -12262,7 +12262,7 @@
 (define-method get_supported_types
   (of-object "GAppInfo")
   (c-name "g_app_info_get_supported_types")
-  (return-type "char**")
+  (return-type "const-char**")
 )
 
 (define-method launch

@@ -12454,7 +12454,7 @@
   (c-name "g_converter_convert")
   (return-type "GConverterResult")
   (parameters
-    '("void*" "inbuf")
+    '("const-void*" "inbuf")
     '("gsize" "inbuf_size")
     '("void*" "outbuf")
     '("gsize" "outbuf_size")

@@ -14974,7 +14974,7 @@
   (c-name "g_pollable_output_stream_write_nonblocking")
   (return-type "gssize")
   (parameters
-    '("void*" "buffer")
+    '("const-void*" "buffer")
     '("gsize" "count")
     '("GCancellable*" "cancellable")
     '("GError**" "error")
