--- gio_vfuncs.defs
+++ gio_vfuncs.defs
@@ -258,7 +258,7 @@
   (return-type "void")
   (parameters
     '("GVariant*" "changed_properties")
+    '("const-gchar*const*" "invalidated_properties")
-    '("const-gchar*-const*" "invalidated_properties")
   )
 )


@@ -791,7 +791,7 @@
   (return-type "void")
   (parameters
     '("const-char*" "message")
+    '("const-gchar**" "choices")
-    '("const-char*" "choices")
   )
 )

@@ -2156,7 +2156,7 @@
   (parameters
     '("const-void*" "inbuf")
     '("gsize" "inbuf_size")
+    '("void*" "outbuf")
-    '("const-void*" "outbuf")
     '("gsize" "outbuf_size")
     '("GConverterFlags" "flags")
     '("gsize*" "bytes_read")

@@ -3920,7 +3920,7 @@
   (of-object "GPollableInputStream")
   (return-type "gssize")
   (parameters
+    '("void*" "buffer")
-    '("const-void*" "buffer")
     '("gsize" "count")
     '("GError**" "error")
   )
