--- ./../../gio/src/gio_enums.defs.orig	2015-07-24 16:57:40.866238234 +0200
+++ ./../../gio/src/gio_enums.defs	2015-07-24 16:57:40.870238424 +0200
@@ -495,10 +495,11 @@
     '("timed-out" "G_IO_ERROR_TIMED_OUT" "24")
     '("would-recurse" "G_IO_ERROR_WOULD_RECURSE" "25")
     '("busy" "G_IO_ERROR_BUSY" "26")
     '("would-block" "G_IO_ERROR_WOULD_BLOCK" "27")
     '("host-not-found" "G_IO_ERROR_HOST_NOT_FOUND" "28")
+    '("host-was-not-found" "G_IO_ERROR_HOST_WAS_NOT_FOUND" "28")
     '("would-merge" "G_IO_ERROR_WOULD_MERGE" "29")
     '("failed-handled" "G_IO_ERROR_FAILED_HANDLED" "30")
     '("too-many-open-files" "G_IO_ERROR_TOO_MANY_OPEN_FILES" "31")
     '("not-initialized" "G_IO_ERROR_NOT_INITIALIZED" "32")
     '("address-in-use" "G_IO_ERROR_ADDRESS_IN_USE" "33")
