--- gio_vfuncs.defs
+++ gio_vfuncs.defs
@@ -258,7 +258,7 @@
   (return-type "void")
   (parameters
     '("GVariant*" "changed_properties")
+    '("const-gchar*const*" "invalidated_properties")
-    '("const-gchar*-const*" "invalidated_properties")
   )
 )


@@ -791,7 +791,7 @@
   (return-type "void")
   (parameters
     '("const-char*" "message")
+    '("const-gchar**" "choices")
-    '("const-char*" "choices")
   )
 )

@@ -2154,7 +2154,7 @@
   (of-object "GConverter")
   (return-type "GConverterResult")
   (parameters
+    '("const-void*" "inbuf")
-    '("void*" "inbuf")
     '("gsize" "inbuf_size")
     '("void*" "outbuf")
     '("gsize" "outbuf_size")

@@ -3948,7 +3948,7 @@
   (of-object "GPollableOutputStream")
   (return-type "gssize")
   (parameters
+    '("const-void*" "buffer")
-    '("void*" "buffer")
     '("gsize" "count")
     '("GError**" "error")
   )
