--- a/gio/src/gio_enums.defs
+++ b/gio/src/gio_enums.defs
@@ -533,6 +533,7 @@
     '("busy" "G_IO_ERROR_BUSY" "26")
     '("would_block" "G_IO_ERROR_WOULD_BLOCK" "27")
     '("host_not_found" "G_IO_ERROR_HOST_NOT_FOUND" "28")
+    '("host-not-found" "G_IO_ERROR_HOST_WAS_NOT_FOUND" "28")
     '("would_merge" "G_IO_ERROR_WOULD_MERGE" "29")
     '("failed_handled" "G_IO_ERROR_FAILED_HANDLED" "30")
     '("too_many_open_files" "G_IO_ERROR_TOO_MANY_OPEN_FILES" "31")
