--- a/gio_methods.defs
+++ b/gio_methods.defs
@@ -1557,7 +1557,7 @@
 (define-method get_environ
   (of-object "GApplicationCommandLine")
   (c-name "g_application_command_line_get_environ")
-  (return-type "gchar**")
+  (return-type "const-gchar*-const*")
 )
 
 (define-method get_exit_status

@@ -1696,7 +1696,7 @@
   (c-name "g_buffered_input_stream_peek")
   (return-type "gsize")
   (parameters
+    '("void*" "buffer")
-    '("const-void*" "buffer")
     '("gsize" "offset")
     '("gsize" "count")
   )

@@ -4406,7 +4406,7 @@
 (define-method list_actions
   (of-object "GDesktopAppInfo")
   (c-name "g_desktop_app_info_list_actions")
-  (return-type "gchar**")
+  (return-type "const-gchar*-const*")
 )
 
 (define-function g_emblem_new

@@ -5713,7 +5713,7 @@
   (c-name "g_inet_address_new_from_bytes")
   (return-type "GInetAddress*")
   (parameters
-    '("guint8*" "bytes")
+    '("const-guint8*" "bytes")
     '("GSocketFamily" "family")
   )
 )

@@ -5985,7 +5985,7 @@
   (c-name "g_input_stream_read")
   (return-type "gssize")
   (parameters
+    '("void*" "buffer")
-    '("const-void*" "buffer")
     '("gsize" "count")
     '("GCancellable*" "cancellable")
     '("GError**" "error")

@@ -5997,7 +5997,7 @@
   (c-name "g_input_stream_read_all")
   (return-type "gboolean")
   (parameters
+    '("void*" "buffer")
-    '("const-void*" "buffer")
     '("gsize" "count")
     '("gsize*" "bytes_read")
     '("GCancellable*" "cancellable")

@@ -8157,7 +8157,7 @@
   (return-type "gboolean")
   (parameters
     '("const-gchar*" "key")
-    '("gchar**" "value")
+    '("const-gchar*-const*" "value")
   )
 )

@@ -8980,7 +8980,7 @@
   (c-name "g_socket_receive")
   (return-type "gssize")
   (parameters
+    '("gchar*" "buffer")
-    '("const-gchar*" "buffer")
     '("gsize" "size")
     '("GCancellable*" "cancellable")
     '("GError**" "error")

@@ -12262,7 +12262,7 @@
 (define-method get_supported_types
   (of-object "GAppInfo")
   (c-name "g_app_info_get_supported_types")
-  (return-type "char**")
+  (return-type "const-char**")
 )
 
 (define-method launch

@@ -12456,7 +12456,7 @@
   (parameters
     '("const-void*" "inbuf")
     '("gsize" "inbuf_size")
+    '("void*" "outbuf")
-    '("const-void*" "outbuf")
     '("gsize" "outbuf_size")
     '("GConverterFlags" "flags")
     '("gsize*" "bytes_read")

@@ -14941,7 +14941,7 @@
   (c-name "g_pollable_input_stream_read_nonblocking")
   (return-type "gssize")
   (parameters
+    '("void*" "buffer")
-    '("const-void*" "buffer")
     '("gsize" "count")
     '("GCancellable*" "cancellable")
     '("GError**" "error")
