--- gio_signals.defs	2011-06-19 18:11:11.000000000 -0400
+++ gio_signals_new.defs	2011-06-19 17:18:06.000000000 -0400
@@ -36,7 +36,7 @@
   (when "last")
   (parameters
     '("const-gchar*" "p0")
-    '("GVariant" "p1")
+    '("GVariant*" "p1")
   )
 )
 
@@ -526,7 +526,7 @@
   (when "last")
   (parameters
     '("const-gchar*" "p0")
-    '("GStrv*" "p1")
+    '("const-gchar**" "p1")
   )
 )
 
@@ -1613,8 +1613,8 @@
   (return-type "void")
   (when "last")
   (parameters
-    '("GVariant" "p0")
-    '("GStrv*" "p1")
+    '("GVariant*" "p0")
+    '("const-gchar*const*" "p1")
   )
 )
 
@@ -1625,7 +1625,7 @@
   (parameters
     '("const-gchar*" "p0")
     '("const-gchar*" "p1")
-    '("GVariant" "p2")
+    '("GVariant*" "p2")
   )
 )
 
