--- ./../../gio/src/gio_signals.defs.orig	2017-09-02 10:48:27.265194882 +0200
+++ ./../../gio/src/gio_signals.defs	2017-09-02 10:48:27.309194987 +0200
@@ -83,11 +83,11 @@
   (of-object "GActionGroup")
   (return-type "void")
   (when "last")
   (parameters
     '("const-gchar*" "p0")
-    '("GVariant" "p1")
+    '("GVariant*" "p1")
   )
 )
 
 ;; From GApplication
 
@@ -734,11 +734,11 @@
   (of-object "GMountOperation")
   (return-type "void")
   (when "last")
   (parameters
     '("const-gchar*" "p0")
-    '("GStrv*" "p1")
+    '("const-gchar**" "p1")
   )
 )
 
 (define-signal reply
   (of-object "GMountOperation")
@@ -1100,11 +1100,11 @@
 (define-signal writable-change-event
   (of-object "GSettings")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("guint" "p0")
+    '("GQuark" "p0")
   )
 )
 
 (define-property settings-schema
   (of-object "GSettings")
@@ -1184,20 +1184,20 @@
 (define-signal activate
   (of-object "GSimpleAction")
   (return-type "void")
   (when "last")
   (parameters
-    '("GVariant" "p0")
+    '("GVariant*" "p0")
   )
 )
 
 (define-signal change-state
   (of-object "GSimpleAction")
   (return-type "void")
   (when "last")
   (parameters
-    '("GVariant" "p0")
+    '("GVariant*" "p0")
   )
 )
 
 (define-property name
   (of-object "GSimpleAction")
@@ -1842,11 +1842,11 @@
 (define-signal writable-change-event
   (of-object "GSettings")
   (return-type "gboolean")
   (when "last")
   (parameters
-    '("guint" "p0")
+    '("GQuark" "p0")
   )
 )
 
 (define-property settings-schema
   (of-object "GSettings")
@@ -2715,23 +2715,23 @@
 (define-signal g-properties-changed
   (of-object "GDBusProxy")
   (return-type "void")
   (when "last")
   (parameters
-    '("GVariant" "p0")
-    '("GStrv*" "p1")
+    '("GVariant*" "p0")
+    '("const-gchar*const*" "p1")
   )
 )
 
 (define-signal g-signal
   (of-object "GDBusProxy")
   (return-type "void")
   (when "last")
   (parameters
     '("const-gchar*" "p0")
     '("const-gchar*" "p1")
-    '("GVariant" "p2")
+    '("GVariant*" "p2")
   )
 )
 
 (define-property g-connection
   (of-object "GDBusProxy")
