--- ./../../gio/src/gio_signals.defs.orig	2017-09-11 09:52:55.272837455 +0200
+++ ./../../gio/src/gio_signals.defs	2017-09-11 14:56:39.662521043 +0200
@@ -87,11 +87,11 @@
   (return-type "void")
   (flags "Run Last, Must Collect")
   (detailed #t)
   (parameters
     '("const-gchar*" "p0")
-    '("GVariant" "p1")
+    '("GVariant*" "p1")
   )
 )
 
 ;; From GApplication
 
@@ -738,11 +738,11 @@
   (of-object "GMountOperation")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("const-gchar*" "p0")
-    '("GStrv*" "p1")
+    '("const-gchar**" "p1")
   )
 )
 
 (define-signal reply
   (of-object "GMountOperation")
@@ -1106,11 +1106,11 @@
 (define-signal writable-change-event
   (of-object "GSettings")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("guint" "p0")
+    '("GQuark" "p0")
   )
 )
 
 (define-property settings-schema
   (of-object "GSettings")
@@ -1190,20 +1190,20 @@
 (define-signal activate
   (of-object "GSimpleAction")
   (return-type "void")
   (flags "Run Last, Must Collect")
   (parameters
-    '("GVariant" "p0")
+    '("GVariant*" "p0")
   )
 )
 
 (define-signal change-state
   (of-object "GSimpleAction")
   (return-type "void")
   (flags "Run Last, Must Collect")
   (parameters
-    '("GVariant" "p0")
+    '("GVariant*" "p0")
   )
 )
 
 (define-property name
   (of-object "GSimpleAction")
@@ -1850,11 +1850,11 @@
 (define-signal writable-change-event
   (of-object "GSettings")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("guint" "p0")
+    '("GQuark" "p0")
   )
 )
 
 (define-property settings-schema
   (of-object "GSettings")
@@ -2723,23 +2723,23 @@
 (define-signal g-properties-changed
   (of-object "GDBusProxy")
   (return-type "void")
   (flags "Run Last, Must Collect")
   (parameters
-    '("GVariant" "p0")
-    '("GStrv*" "p1")
+    '("GVariant*" "p0")
+    '("const-gchar*const*" "p1")
   )
 )
 
 (define-signal g-signal
   (of-object "GDBusProxy")
   (return-type "void")
   (flags "Run Last, Must Collect")
   (parameters
     '("const-gchar*" "p0")
     '("const-gchar*" "p1")
-    '("GVariant" "p2")
+    '("GVariant*" "p2")
   )
 )
 
 (define-property g-connection
   (of-object "GDBusProxy")
