--- tools/gen_scripts/../../gio/src/gio_vfuncs.defs.orig	2026-03-10 09:27:08.610856908 +0100
+++ tools/gen_scripts/../../gio/src/gio_vfuncs.defs	2026-03-09 19:01:47.002316415 +0100
@@ -411,11 +411,11 @@
 
 (define-vfunc convert
   (of-object "GConverter")
   (return-type "GConverterResult")
   (parameters
-   '("void*" "inbuf")
+   '("const-void*" "inbuf")
    '("gsize" "inbuf_size")
    '("void*" "outbuf")
    '("gsize" "outbuf_size")
    '("GConverterFlags" "flags")
    '("gsize*" "bytes_read")
@@ -2314,11 +2314,11 @@
 
 (define-vfunc write_nonblocking
   (of-object "GPollableOutputStream")
   (return-type "gssize")
   (parameters
-   '("void*" "buffer")
+   '("const-void*" "buffer")
    '("gsize" "count")
    '("GError**" "error")
   )
 )
 
@@ -3765,11 +3765,11 @@
 
 (define-vfunc write_fn
   (of-object "GOutputStream")
   (return-type "gssize")
   (parameters
-   '("void*" "buffer")
+   '("const-void*" "buffer")
    '("gsize" "count")
    '("GCancellable*" "cancellable")
    '("GError**" "error")
   )
 )
