--- gio_methods.defs	2011-06-19 18:23:06.000000000 -0400
+++ gio_methods_new.defs	2011-06-19 18:21:21.000000000 -0400
@@ -867,8 +867,8 @@
   (c-name "GFileQueryInfoFlags")
   (gtype-id "G_TYPE_FILE_QUERY_INFO_FLAGS")
   (values
-    '("ne" "G_FILE_QUERY_INFO_NONE")
-    '("follow-symlinks" "G_FILE_QUERY_INFO_NOFOLLOW_SYMLINKS")
+    '("none" "G_FILE_QUERY_INFO_NONE")
+    '("nofollow-symlinks" "G_FILE_QUERY_INFO_NOFOLLOW_SYMLINKS")
   )
 )
 
@@ -888,7 +888,7 @@
   (c-name "GMountMountFlags")
   (gtype-id "G_TYPE_MOUNT_MOUNT_FLAGS")
   (values
-    '("e" "G_MOUNT_MOUNT_NONE")
+    '("none" "G_MOUNT_MOUNT_NONE")
   )
 )
 
@@ -907,7 +907,7 @@
   (c-name "GDriveStartFlags")
   (gtype-id "G_TYPE_DRIVE_START_FLAGS")
   (values
-    '("e" "G_DRIVE_START_NONE")
+    '("none" "G_DRIVE_START_NONE")
   )
 )
 
@@ -1321,8 +1321,8 @@
   (c-name "GDBusCallFlags")
   (gtype-id "G_TYPE_D_BUS_CALL_FLAGS")
   (values
-    '("ne" "G_DBUS_CALL_FLAGS_NONE")
-    '("-auto-start" "G_DBUS_CALL_FLAGS_NO_AUTO_START")
+    '("none" "G_DBUS_CALL_FLAGS_NONE")
+    '("no-auto-start" "G_DBUS_CALL_FLAGS_NO_AUTO_START")
   )
 )
 
@@ -1344,9 +1344,9 @@
   (c-name "GDBusMessageFlags")
   (gtype-id "G_TYPE_D_BUS_MESSAGE_FLAGS")
   (values
-    '("ne" "G_DBUS_MESSAGE_FLAGS_NONE")
-    '("-reply-expected" "G_DBUS_MESSAGE_FLAGS_NO_REPLY_EXPECTED")
-    '("-auto-start" "G_DBUS_MESSAGE_FLAGS_NO_AUTO_START")
+    '("none" "G_DBUS_MESSAGE_FLAGS_NONE")
+    '("no-reply-expected" "G_DBUS_MESSAGE_FLAGS_NO_REPLY_EXPECTED")
+    '("no-auto-start" "G_DBUS_MESSAGE_FLAGS_NO_AUTO_START")
   )
 )
 
@@ -1405,8 +1405,8 @@
   (c-name "GDBusSignalFlags")
   (gtype-id "G_TYPE_D_BUS_SIGNAL_FLAGS")
   (values
-    '("ne" "G_DBUS_SIGNAL_FLAGS_NONE")
-    '("-match-rule" "G_DBUS_SIGNAL_FLAGS_NO_MATCH_RULE")
+    '("none" "G_DBUS_SIGNAL_FLAGS_NONE")
+    '("no-match-rule" "G_DBUS_SIGNAL_FLAGS_NO_MATCH_RULE")
   )
 )
 
