--- gio_signals.defs	2012-02-28 13:19:25.000000000 -0500
+++ gio_signals.defs.new	2012-02-28 12:47:50.000000000 -0500
@@ -83,7 +83,7 @@
   (when "last")
   (parameters
     '("const-gchar*" "p0")
-    '("GVariant" "p1")
+    '("GVariant*" "p1")
   )
 )
 
@@ -579,7 +579,7 @@
   (when "last")
   (parameters
     '("const-gchar*" "p0")
-    '("GStrv*" "p1")
+    '("const-gchar**" "p1")
   )
 )
 
@@ -849,7 +849,7 @@
   (return-type "void")
   (when "last")
   (parameters
-    '("GVariant" "p0")
+    '("GVariant*" "p0")
   )
 )
 
@@ -858,7 +858,7 @@
   (return-type "void")
   (when "last")
   (parameters
-    '("GVariant" "p0")
+    '("GVariant*" "p0")
   )
 )
 
@@ -1824,8 +1824,8 @@
   (return-type "void")
   (when "last")
   (parameters
-    '("GVariant" "p0")
-    '("GStrv*" "p1")
+    '("GVariant*" "p0")
+    '("const-gchar*const*" "p1")
   )
 )
 
@@ -1836,7 +1836,7 @@
   (parameters
     '("const-gchar*" "p0")
     '("const-gchar*" "p1")
-    '("GVariant" "p2")
+    '("GVariant*" "p2")
   )
 )
 
