--- tools/gen_scripts/../../gio/src/gio_signals.defs.orig	2022-04-06 10:10:49.446569490 +0200
+++ tools/gen_scripts/../../gio/src/gio_signals.defs	2022-04-06 10:20:32.616923494 +0200
@@ -87,11 +87,11 @@
   (return-type "void")
   (flags "Run Last, Must Collect")
   (detailed #t)
   (parameters
     '("const-gchar*" "p0")
-    '("GVariant" "p1")
+    '("GVariant*" "p1")
   )
 )
 
 ;; From GApplication
 
@@ -744,11 +744,11 @@
   (of-object "GMountOperation")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("const-gchar*" "p0")
-    '("GStrv*" "p1")
+    '("const-gchar**" "p1")
   )
 )
 
 (define-signal reply
   (of-object "GMountOperation")
@@ -1142,11 +1142,11 @@
 (define-signal writable-change-event
   (of-object "GSettings")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("guint" "p0")
+    '("GQuark" "p0")
   )
 )
 
 (define-property settings-schema
   (of-object "GSettings")
@@ -1226,20 +1226,20 @@
 (define-signal activate
   (of-object "GSimpleAction")
   (return-type "void")
   (flags "Run Last, Must Collect")
   (parameters
-    '("GVariant" "p0")
+    '("GVariant*" "p0")
   )
 )
 
 (define-signal change-state
   (of-object "GSimpleAction")
   (return-type "void")
   (flags "Run Last, Must Collect")
   (parameters
-    '("GVariant" "p0")
+    '("GVariant*" "p0")
   )
 )
 
 (define-property name
   (of-object "GSimpleAction")
@@ -1886,11 +1886,11 @@
 (define-signal writable-change-event
   (of-object "GSettings")
   (return-type "gboolean")
   (flags "Run Last")
   (parameters
-    '("guint" "p0")
+    '("GQuark" "p0")
   )
 )
 
 (define-property settings-schema
   (of-object "GSettings")
@@ -2926,23 +2926,23 @@
   (parameters
     '("GDBusObjectProxy*" "p0")
     '("GDBusProxy*" "p1")
     '("const-gchar*" "p2")
     '("const-gchar*" "p3")
-    '("GVariant" "p4")
+    '("GVariant*" "p4")
   )
 )
 
 (define-signal interface-proxy-properties-changed
   (of-object "GDBusObjectManagerClient")
   (return-type "void")
   (flags "Run Last")
   (parameters
     '("GDBusObjectProxy*" "p0")
     '("GDBusProxy*" "p1")
-    '("GVariant" "p2")
-    '("GStrv*" "p3")
+    '("GVariant*" "p2")
+    '("const-gchar*const*" "p3")
   )
 )
 
 (define-property bus-type
   (of-object "GDBusObjectManagerClient")
@@ -3115,12 +3115,12 @@
 (define-signal g-properties-changed
   (of-object "GDBusProxy")
   (return-type "void")
   (flags "Run Last, Must Collect")
   (parameters
-    '("GVariant" "p0")
-    '("GStrv*" "p1")
+    '("GVariant*" "p0")
+    '("const-gchar*const*" "p1")
   )
 )
 
 (define-signal g-signal
   (of-object "GDBusProxy")
@@ -3128,11 +3128,11 @@
   (flags "Run Last, Must Collect")
   (detailed #t)
   (parameters
     '("const-gchar*" "p0")
     '("const-gchar*" "p1")
-    '("GVariant" "p2")
+    '("GVariant*" "p2")
   )
 )
 
 (define-property g-connection
   (of-object "GDBusProxy")
